Upgrade govuk frontend to v5

kr8n3r opened the pull request:
## What 

Update the app to use govuk-frontend v5.4.0 for WCAG 2.2 compliance.
See individual commits for detailed changes.

Addresses https://trello.com/c/aTQLmIRB/856-bump-design-system-to-v5-in-document-download

![Screenshot 2024-07-11 at 12 35 31](https://github.com/alphagov/document-download-frontend/assets/3758555/b45ea180-6d45-4281-8755-ede4da797d25)
![Screenshot 2024-07-11 at 12 34 57](https://github.com/alphagov/document-download-frontend/assets/3758555/e620b845-e04c-4bca-9f3e-b60648a024f6)
![Screenshot 2024-07-11 at 12 35 18](https://github.com/alphagov/document-download-frontend/assets/3758555/87059966-8942-4af2-82f4-cdce22985859)

---

🚨⚠️ This will be deployed automatically all the way to production when you click merge ⚠️🚨

For more information, including how to check this deployment on preview or staging first before it goes to production, see our [team wiki section on deployment](https://github.com/alphagov/notifications-manuals/wiki/Merging-and-deploying#deployment)

tombye commented on app/assets/javascripts/main.mjs:
```diff
@@ -1,7 +1,15 @@
 
-export {
-  Frontend
+// Find first error summary module to enhance.
```
Should we just use `createAll` to instatiate this module and the `$skipLink`? The guidance seems to say we should only not use it if we intend to handle any errors thrown ourselves, which we don't.

kr8n3r replied:
yep, correct we can just use that. updated

tombye reviewed (changes requested):
I added a comment and we should also stop using the 'element' param with the `govukButton` component because it was deprecated in V5.1.0: https://github.com/alphagov/govuk-frontend/releases/tag/v5.1.0

kr8n3r commented:
> I added a comment and we should also stop using the 'element' param with the `govukButton` component because it was deprecated in V5.1.0: https://github.com/alphagov/govuk-frontend/releases/tag/v5.1.0

create spot! removed from the button

tombye reviewed (approved):
Thanks for doing the fixes. Great, speedy work 👍🏻
